algo/src/main/java/org/neo4j/graphalgo/walking/NodeWalkerProc.java

```diff
@@ -0,0 +1,167 @@
+package org.neo4j.graphalgo.walking;
+
+import org.neo4j.graphalgo.api.Graph;
+import org.neo4j.graphalgo.api.GraphFactory;
+import org.neo4j.graphalgo.core.GraphLoader;
+import org.neo4j.graphalgo.core.ProcedureConfiguration;
+import org.neo4j.graphalgo.core.utils.*;
+import org.neo4j.graphalgo.core.utils.paged.AllocationTracker;
+import org.neo4j.graphalgo.impl.walking.NodeWalker;
+import org.neo4j.graphalgo.impl.walking.WalkPath;
+import org.neo4j.graphalgo.impl.walking.WalkResult;
+import org.neo4j.graphalgo.results.PageRankScore;
+import org.neo4j.graphalgo.results.VirtualRelationship;
+import org.neo4j.graphdb.*;
+import org.neo4j.internal.kernel.api.NodeLabelIndexCursor;
+import org.neo4j.kernel.api.KernelTransaction;
+import org.neo4j.kernel.internal.GraphDatabaseAPI;
+import org.neo4j.logging.Log;
+import org.neo4j.procedure.*;
+
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.stream.*;
+
+public class NodeWalkerProc  {
+
+    private static final RelationshipType NEXT = RelationshipType.withName("NEXT");
+    @Context
+    public GraphDatabaseAPI api;
+
+    @Context
+    public Log log;
+
+    @Context
+    public KernelTransaction transaction;
+
+
+
+    @Procedure(name = "algo.randomWalk.stream", mode = Mode.READ)
+    @Description("CALL algo.randomWalk.stream(label, relationship, start:null=all/[ids]/label, steps, walks, {mode:random/node2vec, return:1.0, inOut:1.0, path:false/true concurrency:4, direction:'BOTH'}) " +
```

Comment:
I wonder if we should make this API the same as we do for the shortestpath procs where you pass in the label + relationship as config parameters. So it would then read:

```
public Stream<WalkResult> randomWalk(
             @Name(value = "start", defaultValue = "null") Object start,
             @Name(value = "steps", defaultValue = "10") long steps,
             @Name(value = "walks", defaultValue = "1") long walks,
             @Name(value = "config", defaultValue = "{}") Map<String, Object> config) {

String label = config.get("nodeQuery");
String relationshipType = config.get("relationshipQuery")
....
```

I think that would be easier to understand as an API. We could perhaps also create a config parameter to go with the `start` parameter where you could indicate what you're intending to pass in as your starting point.

So it could default to 'label' but you could have other things too:

```
CALL algo.randomWalk.stream(node,....{startType: "node"})
CALL algo.randomWalk.stream(1,....{startType: "nodeId"})
```

Would save us having to do the type check on the value and we do a similar thing when allowing people to pass in label + reltype vs Cypher queries for graph projections.